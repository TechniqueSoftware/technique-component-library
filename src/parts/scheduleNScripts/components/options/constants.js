export const SCRIPT_CONTACT_TYPES = {
  EMAIL: 2,
  SMS: 3,
  NONE: 4
}
