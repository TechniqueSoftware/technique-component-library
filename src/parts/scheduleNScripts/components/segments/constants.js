export const FOLLOWUP_STATUS = {
  APPT_BOOKED: 2,
  MEMBER: 6,
  WEB_LEAD: 11
}
