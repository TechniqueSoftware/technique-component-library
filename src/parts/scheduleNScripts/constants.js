export const TP_ACTION_TYPES = {
  CALL: 1,
  EMAIL: 2,
  TEXT: 3
}

export const SCRIPT_TABNAME = [{
  id: '1',
  name: 'Call',
  className: 'BsTelephone'
}, {
  id: '2',
  name: 'Email',
  className: 'BsEnvelope'
}, {
  id: '3',
  name: 'Text',
  className: 'BsChatDots'
}]

export const SEGMENT_TABNAME = [
  {
    id: '0',
    name: 'Manage Segments'
  },
  {
    id: '1',
    name: 'Copy Segments'
  }
]

export const SNACKBAR_MESSAGES = {
  NEW_SUCCESS: 'Touchpoint has been created successfully!',
  EDIT_SUCCESS: 'Touchpoint has been updated!',
  DELETE_SUCCESS: 'Touchpoint has been deleted successfully!',
  SCRIPT_SUCCESS: 'Script has been saved successfully!',
  SEGMENT_NEW: 'Segment has been created successfully!',
  SEGMENT_EDIT: 'Changes for this segment have been saved successfully!',
  SEGMENT_DELETE: 'Segment has been deleted successfully!'
}

export const STATUSES_WITH_SEGMENTS = [
  { type: 1, id: 11, attributeType: 'Marketing Source', attributeId: 'marketing' }, // Prospecting -> Web Lead
  { type: 1, id: 2, attributeType: 'Marketing Source', attributeId: 'marketing' }, // Prospecting -> Appt Booked
  { type: 1, id: 6, attributeType: 'Membership Type', attributeId: 'membership' } // Member -> Member
]

export const PERMISSION_KEYS = {
  SCHEDULES: {
    1: 'canEditMemberSalesSchedules',
    2: 'canEditPtSchedules',
    3: 'canEditStudioSchedules'
  },
  SCRIPTS: {
    1: 'canEditMemberSalesScripts',
    2: 'canEditPtScripts',
    3: 'canEditStudioScripts'
  }
}
