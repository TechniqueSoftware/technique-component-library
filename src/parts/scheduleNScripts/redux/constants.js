export const TP_OPTIONS = {
  MENU: 'menu',
  NEW: 'new',
  EDIT: 'edit'
}

export const GROUP_NAMES = {
  PROSPECT: 'Prospect',
  ACTIVE_MEMBER: 'Active Member',
  INACTIVE_MEMBER: 'Inactive Member',
  PT_PROSPECT: 'PT Prospect',
  PT_MEMBER: 'PT Member',
  PT_OPT_OUT: 'PT Opt Out',
  PERSONAL_TRAINING: 'Personal Training'
}
