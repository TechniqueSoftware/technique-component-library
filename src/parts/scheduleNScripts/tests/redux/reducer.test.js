import { TYPES, REDUCER } from '../../redux'

import { TP_OPTIONS } from '../../redux/constants'

import { followupsMock, segmentsMock } from './mock'

describe('REDUCER', () => {
  describe('Followups', () => {
    it('gets', () => {
      expect(REDUCER(undefined, { type: TYPES.FOLLOWUPS.GET })).toMatchSnapshot()
    })

    it('errors', () => {
      expect(
        REDUCER(undefined, {
          type: TYPES.FOLLOWUPS.ERROR,
          payload: 'some error here'
        })
      ).toMatchSnapshot()
    })

    it('succeeds', () => {
      expect(
        REDUCER(undefined, {
          type: TYPES.FOLLOWUPS.SUCCESS,
          payload: followupsMock.content[0]
        })
      ).toMatchSnapshot()
    })

    it('sets permission', () => {
      expect(
        REDUCER(undefined, {
          type: TYPES.FOLLOWUPS.PERMISSION,
          payload: {
            canEditSchedules: true,
            canEditScripts: false
          }
        })
      ).toMatchSnapshot()
    })

    it('sets', () => {
      expect(
        REDUCER(undefined, {
          type: TYPES.FOLLOWUPS.SET,
          payload: followupsMock.content[0]
        })
      ).toMatchSnapshot()
    })
  })

  describe('Touchpoints', () => {
    it('opens', () => {
      expect(
        REDUCER(undefined, {
          type: TYPES.TOUCHPOINT.OPEN,
          payload: { action: TP_OPTIONS.MENU, data: { b: 1 } }
        })
      ).toMatchSnapshot()
    })

    it('closes', () => {
      expect(
        REDUCER(undefined, {
          type: TYPES.TOUCHPOINT.CLOSE
        })
      ).toMatchSnapshot()
    })

    it('new', () => {
      expect(
        REDUCER(undefined, {
          type: TYPES.TOUCHPOINT.NEW,
          payload: { action: TP_OPTIONS.NEW, data: { a: 1 } }
        })
      ).toMatchSnapshot()
    })

    it('new success', () => {
      expect(
        REDUCER(undefined, {
          type: TYPES.TOUCHPOINT.NEW_SUCCESS,
          payload: { a: 1, b: 2 }
        })
      ).toMatchSnapshot()
    })

    it('new error', () => {
      expect(
        REDUCER(undefined, {
          type: TYPES.TOUCHPOINT.NEW_ERROR,
          payload: 'There has been an error'
        })
      ).toMatchSnapshot()
    })

    it('edit', () => {
      expect(
        REDUCER(undefined, {
          type: TYPES.TOUCHPOINT.EDIT
        })
      ).toMatchSnapshot()
    })

    it('edit touchpoint', () => {
      expect(
        REDUCER(undefined, {
          type: TYPES.TOUCHPOINT.EDIT_TOUCHPOINT
        })
      ).toMatchSnapshot()
    })

    it('edit touchpoint success', () => {
      expect(
        REDUCER(undefined, {
          type: TYPES.TOUCHPOINT.EDIT_SUCCESS,
          payload: { followups: [], snackbar: 'EDIT_SUCCESS' }
        })
      ).toMatchSnapshot()
    })

    it('edit error', () => {
      expect(
        REDUCER(undefined, {
          type: TYPES.TOUCHPOINT.EDIT_ERROR,
          payload: 'There has been an error'
        })
      ).toMatchSnapshot()
    })

    it('delete', () => {
      expect(
        REDUCER(undefined, {
          type: TYPES.TOUCHPOINT.DELETE
        })
      ).toMatchSnapshot()
    })

    it('delete success', () => {
      expect(
        REDUCER(undefined, {
          type: TYPES.TOUCHPOINT.DELETE_SUCCESS,
          payload: { a: 1, b: 2 }
        })
      ).toMatchSnapshot()
    })

    it('delete error', () => {
      expect(
        REDUCER(undefined, {
          type: TYPES.TOUCHPOINT.DELETE_ERROR,
          payload: 'There has been an error'
        })
      ).toMatchSnapshot()
    })
  })

  describe('Inner permissions', () => {
    it('sets inner permissions', () => {
      expect(REDUCER(undefined, {
        type: TYPES.FOLLOWUPS.SET_INNER_PERMISSIONS,
        payload: { automaticFollowUpAction: false, followUpSegmentsAction: true }
      })
      ).toMatchSnapshot()
    })
  })

  describe('Scripts', () => {
    it('Edit Scripts open', () => {
      expect(
        REDUCER(undefined, {
          type: TYPES.SCRIPTS.EDIT,
          payload: { a: 1, b: 2 }
        })
      ).toMatchSnapshot()
    })

    it('Save Scripts ', () => {
      expect(
        REDUCER(undefined, {
          type: TYPES.SCRIPTS.SAVE,
          payload: { a: 1, b: 2 }
        })
      ).toMatchSnapshot()
    })
  })

  describe('Snackbar', () => {
    it('sets snackbar', () => {
      expect(
        REDUCER(undefined, {
          type: TYPES.SNACKBAR.SET,
          payload: 'Snackbar Msg'
        })
      ).toMatchSnapshot()
    })

    it('closes snackbar', () => {
      expect(
        REDUCER(undefined, {
          type: TYPES.SNACKBAR.CLOSE
        })
      ).toMatchSnapshot()
    })
  })

  describe('Segments', () => {
    it('sets segments', () => {
      expect(
        REDUCER(undefined, {
          type: TYPES.SEGMENTS.SET,
          payload: segmentsMock
        })
      ).toMatchSnapshot()
    })

    it('sets segment attributes', () => {
      expect(
        REDUCER(undefined, {
          type: TYPES.SEGMENTS.SET_ATTRIBUTES,
          payload: { marketing: ['a'], membership: ['b'] }
        })
      ).toMatchSnapshot()
    })

    it('sets segment data', () => {
      expect(
        REDUCER(undefined, {
          type: TYPES.SEGMENTS.SET_DATA,
          payload: { title: 'title', followUpType: 1, followUpStatus: '11' }
        })
      ).toMatchSnapshot()
    })

    it('closes segment modal', () => {
      expect(
        REDUCER(undefined, {
          type: TYPES.SEGMENTS.CLOSE
        })
      ).toMatchSnapshot()
    })

    it('sets segment error', () => {
      expect(
        REDUCER(undefined, {
          type: TYPES.SEGMENTS.ERROR,
          payload: 'Some error here'
        })
      ).toMatchSnapshot()
    })
  })
})
