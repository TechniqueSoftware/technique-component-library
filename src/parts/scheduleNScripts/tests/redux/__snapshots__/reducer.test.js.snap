// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`REDUCER Followups errors 1`] = `
Object {
  "canEditSchedules": false,
  "canEditScripts": false,
  "copyInProcess": Object {},
  "error": "some error here",
  "followups": Array [],
  "innerPermissions": Object {
    "automaticFollowUpAction": false,
    "followUpSegmentsAction": false,
  },
  "loading": false,
  "popupAlert": "",
  "script": Object {},
  "segmentAttributes": Object {
    "marketing": Array [],
    "membership": Array [],
  },
  "segmentList": Array [],
  "segmentLocations": Array [],
  "segmentModalData": Object {
    "error": "",
    "isOpen": false,
    "title": "",
  },
  "segments": Array [],
  "snackbar": "",
  "touchpointCrud": Object {
    "action": "",
    "data": Object {},
    "loading": false,
  },
}
`;

exports[`REDUCER Followups gets 1`] = `
Object {
  "canEditSchedules": false,
  "canEditScripts": false,
  "copyInProcess": Object {},
  "error": "",
  "followups": Array [],
  "innerPermissions": Object {
    "automaticFollowUpAction": false,
    "followUpSegmentsAction": false,
  },
  "loading": true,
  "popupAlert": "",
  "script": Object {},
  "segmentAttributes": Object {
    "marketing": Array [],
    "membership": Array [],
  },
  "segmentList": Array [],
  "segmentLocations": Array [],
  "segmentModalData": Object {
    "error": "",
    "isOpen": false,
    "title": "",
  },
  "segments": Array [],
  "snackbar": "",
  "touchpointCrud": Object {
    "action": "",
    "data": Object {},
    "loading": false,
  },
}
`;

exports[`REDUCER Followups sets 1`] = `
Object {
  "canEditSchedules": false,
  "canEditScripts": false,
  "copyInProcess": Object {},
  "error": "",
  "followups": Object {
    "clubId": 659,
    "followUpTypes": Array [
      Object {
        "followUpStatuses": Array [
          Object {
            "customClubLabel": null,
            "groupName": "Prospect",
            "hidden": "false",
            "id": "4",
            "name": "Active Guest",
          },
          Object {
            "customClubLabel": null,
            "groupName": "Prospect",
            "hidden": "false",
            "id": "5",
            "name": "Expired Guest",
          },
        ],
        "id": "1",
        "name": "Membership",
      },
      Object {
        "followUpStatuses": Array [
          Object {
            "customClubLabel": null,
            "groupName": "PT Prospect",
            "hidden": "false",
            "id": "1",
            "name": "Not Booked",
          },
        ],
        "id": "2",
        "name": "Personal Training",
      },
    ],
  },
  "innerPermissions": Object {
    "automaticFollowUpAction": false,
    "followUpSegmentsAction": false,
  },
  "loading": false,
  "popupAlert": "",
  "script": Object {},
  "segmentAttributes": Object {
    "marketing": Array [],
    "membership": Array [],
  },
  "segmentList": Array [],
  "segmentLocations": Array [],
  "segmentModalData": Object {
    "error": "",
    "isOpen": false,
    "title": "",
  },
  "segments": Array [],
  "snackbar": "",
  "touchpointCrud": Object {
    "action": "",
    "data": Object {},
    "loading": false,
  },
}
`;

exports[`REDUCER Followups sets permission 1`] = `
Object {
  "canEditSchedules": true,
  "canEditScripts": false,
  "copyInProcess": Object {},
  "error": "",
  "followups": Array [],
  "innerPermissions": Object {
    "automaticFollowUpAction": false,
    "followUpSegmentsAction": false,
  },
  "loading": false,
  "popupAlert": "",
  "script": Object {},
  "segmentAttributes": Object {
    "marketing": Array [],
    "membership": Array [],
  },
  "segmentList": Array [],
  "segmentLocations": Array [],
  "segmentModalData": Object {
    "error": "",
    "isOpen": false,
    "title": "",
  },
  "segments": Array [],
  "snackbar": "",
  "touchpointCrud": Object {
    "action": "",
    "data": Object {},
    "loading": false,
  },
}
`;

exports[`REDUCER Followups succeeds 1`] = `
Object {
  "canEditSchedules": undefined,
  "canEditScripts": undefined,
  "copyInProcess": Object {},
  "error": "",
  "followups": undefined,
  "innerPermissions": Object {
    "automaticFollowUpAction": false,
    "followUpSegmentsAction": false,
  },
  "loading": false,
  "popupAlert": "",
  "script": Object {},
  "segmentAttributes": Object {
    "marketing": Array [],
    "membership": Array [],
  },
  "segmentList": Array [],
  "segmentLocations": Array [],
  "segmentModalData": Object {
    "error": "",
    "isOpen": false,
    "title": "",
  },
  "segments": Array [],
  "snackbar": "",
  "touchpointCrud": Object {
    "action": "",
    "data": Object {},
    "loading": false,
  },
}
`;

exports[`REDUCER Inner permissions sets inner permissions 1`] = `
Object {
  "canEditSchedules": false,
  "canEditScripts": false,
  "copyInProcess": Object {},
  "error": "",
  "followups": Array [],
  "innerPermissions": Object {
    "automaticFollowUpAction": false,
    "followUpSegmentsAction": true,
  },
  "loading": false,
  "popupAlert": "",
  "script": Object {},
  "segmentAttributes": Object {
    "marketing": Array [],
    "membership": Array [],
  },
  "segmentList": Array [],
  "segmentLocations": Array [],
  "segmentModalData": Object {
    "error": "",
    "isOpen": false,
    "title": "",
  },
  "segments": Array [],
  "snackbar": "",
  "touchpointCrud": Object {
    "action": "",
    "data": Object {},
    "loading": false,
  },
}
`;

exports[`REDUCER Scripts Edit Scripts open 1`] = `
Object {
  "canEditSchedules": false,
  "canEditScripts": false,
  "copyInProcess": Object {},
  "error": "",
  "followups": Array [],
  "innerPermissions": Object {
    "automaticFollowUpAction": false,
    "followUpSegmentsAction": false,
  },
  "loading": false,
  "popupAlert": "",
  "script": Object {},
  "segmentAttributes": Object {
    "marketing": Array [],
    "membership": Array [],
  },
  "segmentList": Array [],
  "segmentLocations": Array [],
  "segmentModalData": Object {
    "error": "",
    "isOpen": false,
    "title": "",
  },
  "segments": Array [],
  "snackbar": "",
  "touchpointCrud": Object {
    "action": undefined,
    "data": undefined,
  },
}
`;

exports[`REDUCER Scripts Save Scripts  1`] = `
Object {
  "canEditSchedules": false,
  "canEditScripts": false,
  "copyInProcess": Object {},
  "error": "",
  "followups": Array [],
  "innerPermissions": Object {
    "automaticFollowUpAction": false,
    "followUpSegmentsAction": false,
  },
  "loading": false,
  "popupAlert": "",
  "script": Object {
    "data": undefined,
  },
  "segmentAttributes": Object {
    "marketing": Array [],
    "membership": Array [],
  },
  "segmentList": Array [],
  "segmentLocations": Array [],
  "segmentModalData": Object {
    "error": "",
    "isOpen": false,
    "title": "",
  },
  "segments": Array [],
  "snackbar": "",
  "touchpointCrud": Object {
    "action": "",
    "data": Object {},
    "loading": false,
  },
}
`;

exports[`REDUCER Segments closes segment modal 1`] = `
Object {
  "canEditSchedules": false,
  "canEditScripts": false,
  "copyInProcess": Object {},
  "error": "",
  "followups": Array [],
  "innerPermissions": Object {
    "automaticFollowUpAction": false,
    "followUpSegmentsAction": false,
  },
  "loading": false,
  "popupAlert": "",
  "script": Object {},
  "segmentAttributes": Object {
    "marketing": Array [],
    "membership": Array [],
  },
  "segmentList": Array [],
  "segmentLocations": Array [],
  "segmentModalData": Object {
    "error": "",
    "isOpen": false,
    "title": "",
  },
  "segments": Array [],
  "snackbar": "",
  "touchpointCrud": Object {
    "action": "",
    "data": Object {},
    "loading": false,
  },
}
`;

exports[`REDUCER Segments sets segment attributes 1`] = `
Object {
  "canEditSchedules": false,
  "canEditScripts": false,
  "copyInProcess": Object {},
  "error": "",
  "followups": Array [],
  "innerPermissions": Object {
    "automaticFollowUpAction": false,
    "followUpSegmentsAction": false,
  },
  "loading": false,
  "popupAlert": "",
  "script": Object {},
  "segmentAttributes": Object {
    "marketing": Array [
      "a",
    ],
    "membership": Array [
      "b",
    ],
  },
  "segmentList": Array [],
  "segmentLocations": Array [],
  "segmentModalData": Object {
    "error": "",
    "isOpen": false,
    "title": "",
  },
  "segments": Array [],
  "snackbar": "",
  "touchpointCrud": Object {
    "action": "",
    "data": Object {},
    "loading": false,
  },
}
`;

exports[`REDUCER Segments sets segment data 1`] = `
Object {
  "canEditSchedules": false,
  "canEditScripts": false,
  "copyInProcess": Object {},
  "error": "",
  "followups": Array [],
  "innerPermissions": Object {
    "automaticFollowUpAction": false,
    "followUpSegmentsAction": false,
  },
  "loading": false,
  "popupAlert": "",
  "script": Object {},
  "segmentAttributes": Object {
    "marketing": Array [],
    "membership": Array [],
  },
  "segmentList": Array [],
  "segmentLocations": Array [],
  "segmentModalData": Object {
    "error": "",
    "followUpStatus": "11",
    "followUpType": 1,
    "isOpen": true,
    "title": "title",
  },
  "segments": Array [],
  "snackbar": "",
  "touchpointCrud": Object {
    "action": "",
    "data": Object {},
    "loading": false,
  },
}
`;

exports[`REDUCER Segments sets segment error 1`] = `
Object {
  "canEditSchedules": false,
  "canEditScripts": false,
  "copyInProcess": Object {},
  "error": "",
  "followups": Array [],
  "innerPermissions": Object {
    "automaticFollowUpAction": false,
    "followUpSegmentsAction": false,
  },
  "loading": false,
  "popupAlert": "",
  "script": Object {},
  "segmentAttributes": Object {
    "marketing": Array [],
    "membership": Array [],
  },
  "segmentList": Array [],
  "segmentLocations": Array [],
  "segmentModalData": Object {
    "error": "Some error here",
    "isOpen": false,
    "title": "",
  },
  "segments": Array [],
  "snackbar": "",
  "touchpointCrud": Object {
    "action": "",
    "data": Object {},
    "loading": false,
  },
}
`;

exports[`REDUCER Segments sets segments 1`] = `
Object {
  "canEditSchedules": false,
  "canEditScripts": false,
  "copyInProcess": Object {},
  "error": "",
  "followups": Array [],
  "innerPermissions": Object {
    "automaticFollowUpAction": false,
    "followUpSegmentsAction": false,
  },
  "loading": false,
  "popupAlert": "",
  "script": Object {},
  "segmentAttributes": Object {
    "marketing": Array [],
    "membership": Array [],
  },
  "segmentList": Array [],
  "segmentLocations": Array [],
  "segmentModalData": Object {
    "error": "",
    "isOpen": false,
    "title": "",
  },
  "segments": Array [
    Object {
      "createdAt": "2022-11-03T21:47:51Z",
      "createdBy": 20,
      "isDeleted": false,
      "name": "Test Segment",
      "segmentAttributes": Array [
        Object {
          "attributeId": 1936,
          "attributeType": "Marketing Source",
          "attributeValue": "Flyer",
        },
        Object {
          "attributeId": 1940,
          "attributeType": "Marketing Source",
          "attributeValue": "Other",
        },
      ],
      "segmentId": 1935,
      "updatedAt": "2022-11-03T22:01:10Z",
      "updatedBy": 20,
    },
    Object {
      "createdAt": "2022-11-07T16:58:04Z",
      "createdBy": 20,
      "isDeleted": false,
      "name": "Test Segment I",
      "segmentAttributes": Array [
        Object {
          "attributeId": 1944,
          "attributeType": "Marketing Source",
          "attributeValue": "Drive By",
        },
      ],
      "segmentId": 1943,
      "updatedAt": "2022-11-07T16:58:04Z",
      "updatedBy": 20,
    },
  ],
  "snackbar": "",
  "touchpointCrud": Object {
    "action": "",
    "data": Object {},
    "loading": false,
  },
}
`;

exports[`REDUCER Snackbar closes snackbar 1`] = `
Object {
  "canEditSchedules": false,
  "canEditScripts": false,
  "copyInProcess": Object {},
  "error": "",
  "followups": Array [],
  "innerPermissions": Object {
    "automaticFollowUpAction": false,
    "followUpSegmentsAction": false,
  },
  "loading": false,
  "popupAlert": "",
  "script": Object {},
  "segmentAttributes": Object {
    "marketing": Array [],
    "membership": Array [],
  },
  "segmentList": Array [],
  "segmentLocations": Array [],
  "segmentModalData": Object {
    "error": "",
    "isOpen": false,
    "title": "",
  },
  "segments": Array [],
  "snackbar": "",
  "touchpointCrud": Object {
    "action": "",
    "data": Object {},
    "loading": false,
  },
}
`;

exports[`REDUCER Snackbar sets snackbar 1`] = `
Object {
  "canEditSchedules": false,
  "canEditScripts": false,
  "copyInProcess": Object {},
  "error": "",
  "followups": Array [],
  "innerPermissions": Object {
    "automaticFollowUpAction": false,
    "followUpSegmentsAction": false,
  },
  "loading": false,
  "popupAlert": "",
  "script": Object {},
  "segmentAttributes": Object {
    "marketing": Array [],
    "membership": Array [],
  },
  "segmentList": Array [],
  "segmentLocations": Array [],
  "segmentModalData": Object {
    "error": "",
    "isOpen": false,
    "title": "",
  },
  "segments": Array [],
  "snackbar": "Snackbar Msg",
  "touchpointCrud": Object {
    "action": "",
    "data": Object {},
    "loading": false,
  },
}
`;

exports[`REDUCER Touchpoints closes 1`] = `
Object {
  "canEditSchedules": false,
  "canEditScripts": false,
  "copyInProcess": Object {},
  "error": "",
  "followups": Array [],
  "innerPermissions": Object {
    "automaticFollowUpAction": false,
    "followUpSegmentsAction": false,
  },
  "loading": false,
  "popupAlert": "",
  "script": Object {},
  "segmentAttributes": Object {
    "marketing": Array [],
    "membership": Array [],
  },
  "segmentList": Array [],
  "segmentLocations": Array [],
  "segmentModalData": Object {
    "error": "",
    "isOpen": false,
    "title": "",
  },
  "segments": Array [],
  "snackbar": "",
  "touchpointCrud": Object {
    "action": "",
    "data": Object {},
    "loading": false,
  },
}
`;

exports[`REDUCER Touchpoints delete 1`] = `
Object {
  "canEditSchedules": false,
  "canEditScripts": false,
  "copyInProcess": Object {},
  "error": "",
  "followups": Array [],
  "innerPermissions": Object {
    "automaticFollowUpAction": false,
    "followUpSegmentsAction": false,
  },
  "loading": false,
  "popupAlert": "",
  "script": Object {},
  "segmentAttributes": Object {
    "marketing": Array [],
    "membership": Array [],
  },
  "segmentList": Array [],
  "segmentLocations": Array [],
  "segmentModalData": Object {
    "error": "",
    "isOpen": false,
    "title": "",
  },
  "segments": Array [],
  "snackbar": "",
  "touchpointCrud": Object {
    "action": "",
    "data": Object {},
    "loading": true,
  },
}
`;

exports[`REDUCER Touchpoints delete error 1`] = `
Object {
  "canEditSchedules": false,
  "canEditScripts": false,
  "copyInProcess": Object {},
  "error": "",
  "followups": Array [],
  "innerPermissions": Object {
    "automaticFollowUpAction": false,
    "followUpSegmentsAction": false,
  },
  "loading": false,
  "popupAlert": "",
  "script": Object {},
  "segmentAttributes": Object {
    "marketing": Array [],
    "membership": Array [],
  },
  "segmentList": Array [],
  "segmentLocations": Array [],
  "segmentModalData": Object {
    "error": "",
    "isOpen": false,
    "title": "",
  },
  "segments": Array [],
  "snackbar": "",
  "touchpointCrud": Object {
    "action": "",
    "data": Object {},
    "error": "There has been an error",
    "loading": false,
  },
}
`;

exports[`REDUCER Touchpoints delete success 1`] = `
Object {
  "canEditSchedules": false,
  "canEditScripts": false,
  "copyInProcess": Object {},
  "error": "",
  "followups": Object {
    "a": 1,
    "b": 2,
  },
  "innerPermissions": Object {
    "automaticFollowUpAction": false,
    "followUpSegmentsAction": false,
  },
  "loading": false,
  "popupAlert": "",
  "script": Object {},
  "segmentAttributes": Object {
    "marketing": Array [],
    "membership": Array [],
  },
  "segmentList": Array [],
  "segmentLocations": Array [],
  "segmentModalData": Object {
    "error": "",
    "isOpen": false,
    "title": "",
  },
  "segments": Array [],
  "snackbar": "DELETE_SUCCESS",
  "touchpointCrud": Object {
    "action": "",
    "data": Object {},
    "loading": false,
  },
}
`;

exports[`REDUCER Touchpoints edit 1`] = `
Object {
  "canEditSchedules": false,
  "canEditScripts": false,
  "copyInProcess": Object {},
  "error": "",
  "followups": Array [],
  "innerPermissions": Object {
    "automaticFollowUpAction": false,
    "followUpSegmentsAction": false,
  },
  "loading": false,
  "popupAlert": "",
  "script": Object {},
  "segmentAttributes": Object {
    "marketing": Array [],
    "membership": Array [],
  },
  "segmentList": Array [],
  "segmentLocations": Array [],
  "segmentModalData": Object {
    "error": "",
    "isOpen": false,
    "title": "",
  },
  "segments": Array [],
  "snackbar": "",
  "touchpointCrud": Object {
    "action": "edit",
    "data": Object {},
    "loading": false,
  },
}
`;

exports[`REDUCER Touchpoints edit error 1`] = `
Object {
  "canEditSchedules": false,
  "canEditScripts": false,
  "copyInProcess": Object {},
  "error": "",
  "followups": Array [],
  "innerPermissions": Object {
    "automaticFollowUpAction": false,
    "followUpSegmentsAction": false,
  },
  "loading": false,
  "popupAlert": "",
  "script": Object {},
  "segmentAttributes": Object {
    "marketing": Array [],
    "membership": Array [],
  },
  "segmentList": Array [],
  "segmentLocations": Array [],
  "segmentModalData": Object {
    "error": "",
    "isOpen": false,
    "title": "",
  },
  "segments": Array [],
  "snackbar": "",
  "touchpointCrud": Object {
    "action": "",
    "data": Object {},
    "error": "There has been an error",
    "loading": false,
  },
}
`;

exports[`REDUCER Touchpoints edit touchpoint 1`] = `
Object {
  "canEditSchedules": false,
  "canEditScripts": false,
  "copyInProcess": Object {},
  "error": "",
  "followups": Array [],
  "innerPermissions": Object {
    "automaticFollowUpAction": false,
    "followUpSegmentsAction": false,
  },
  "loading": false,
  "popupAlert": "",
  "script": Object {},
  "segmentAttributes": Object {
    "marketing": Array [],
    "membership": Array [],
  },
  "segmentList": Array [],
  "segmentLocations": Array [],
  "segmentModalData": Object {
    "error": "",
    "isOpen": false,
    "title": "",
  },
  "segments": Array [],
  "snackbar": "",
  "touchpointCrud": Object {
    "action": "",
    "data": Object {},
    "loading": true,
  },
}
`;

exports[`REDUCER Touchpoints edit touchpoint success 1`] = `
Object {
  "canEditSchedules": false,
  "canEditScripts": false,
  "copyInProcess": Object {},
  "error": "",
  "followups": Array [],
  "innerPermissions": Object {
    "automaticFollowUpAction": false,
    "followUpSegmentsAction": false,
  },
  "loading": false,
  "popupAlert": "",
  "script": Object {},
  "segmentAttributes": Object {
    "marketing": Array [],
    "membership": Array [],
  },
  "segmentList": Array [],
  "segmentLocations": Array [],
  "segmentModalData": Object {
    "error": "",
    "isOpen": false,
    "title": "",
  },
  "segments": Array [],
  "snackbar": "EDIT_SUCCESS",
  "touchpointCrud": Object {
    "action": "",
    "data": Object {},
    "loading": false,
  },
}
`;

exports[`REDUCER Touchpoints new 1`] = `
Object {
  "canEditSchedules": false,
  "canEditScripts": false,
  "copyInProcess": Object {},
  "error": "",
  "followups": Array [],
  "innerPermissions": Object {
    "automaticFollowUpAction": false,
    "followUpSegmentsAction": false,
  },
  "loading": false,
  "popupAlert": "",
  "script": Object {},
  "segmentAttributes": Object {
    "marketing": Array [],
    "membership": Array [],
  },
  "segmentList": Array [],
  "segmentLocations": Array [],
  "segmentModalData": Object {
    "error": "",
    "isOpen": false,
    "title": "",
  },
  "segments": Array [],
  "snackbar": "",
  "touchpointCrud": Object {
    "action": "",
    "data": Object {},
    "loading": true,
  },
}
`;

exports[`REDUCER Touchpoints new error 1`] = `
Object {
  "canEditSchedules": false,
  "canEditScripts": false,
  "copyInProcess": Object {},
  "error": "",
  "followups": Array [],
  "innerPermissions": Object {
    "automaticFollowUpAction": false,
    "followUpSegmentsAction": false,
  },
  "loading": false,
  "popupAlert": "",
  "script": Object {},
  "segmentAttributes": Object {
    "marketing": Array [],
    "membership": Array [],
  },
  "segmentList": Array [],
  "segmentLocations": Array [],
  "segmentModalData": Object {
    "error": "",
    "isOpen": false,
    "title": "",
  },
  "segments": Array [],
  "snackbar": "",
  "touchpointCrud": Object {
    "action": "",
    "data": Object {},
    "error": "There has been an error",
    "loading": false,
  },
}
`;

exports[`REDUCER Touchpoints new success 1`] = `
Object {
  "canEditSchedules": false,
  "canEditScripts": false,
  "copyInProcess": Object {},
  "error": "",
  "followups": Object {
    "a": 1,
    "b": 2,
  },
  "innerPermissions": Object {
    "automaticFollowUpAction": false,
    "followUpSegmentsAction": false,
  },
  "loading": false,
  "popupAlert": "",
  "script": Object {},
  "segmentAttributes": Object {
    "marketing": Array [],
    "membership": Array [],
  },
  "segmentList": Array [],
  "segmentLocations": Array [],
  "segmentModalData": Object {
    "error": "",
    "isOpen": false,
    "title": "",
  },
  "segments": Array [],
  "snackbar": "NEW_SUCCESS",
  "touchpointCrud": Object {
    "action": "",
    "data": Object {},
    "loading": false,
  },
}
`;

exports[`REDUCER Touchpoints opens 1`] = `
Object {
  "canEditSchedules": false,
  "canEditScripts": false,
  "copyInProcess": Object {},
  "error": "",
  "followups": Array [],
  "innerPermissions": Object {
    "automaticFollowUpAction": false,
    "followUpSegmentsAction": false,
  },
  "loading": false,
  "popupAlert": "",
  "script": Object {},
  "segmentAttributes": Object {
    "marketing": Array [],
    "membership": Array [],
  },
  "segmentList": Array [],
  "segmentLocations": Array [],
  "segmentModalData": Object {
    "error": "",
    "isOpen": false,
    "title": "",
  },
  "segments": Array [],
  "snackbar": "",
  "touchpointCrud": Object {
    "action": "menu",
    "data": Object {
      "b": 1,
    },
  },
}
`;
